components/testmonitor/sources/common/base.mjs

```diff
@@ -0,0 +1,87 @@
+import { DEFAULT_POLLING_SOURCE_TIMER_INTERVAL } from "@pipedream/platform";
+import moment from "moment";
+import testmonitor from "../../testmonitor.app.mjs";
+
+export default {
+  props: {
+    testmonitor,
+    projectId: {
+      propDefinition: [
+        testmonitor,
+        "projectId",
+      ],
+    },
+    db: "$.service.db",
+    timer: {
+      label: "Polling interval",
+      description: "Pipedream will poll the TestMonitor API on this schedule",
+      type: "$.interface.timer",
+      default: {
+        intervalSeconds: DEFAULT_POLLING_SOURCE_TIMER_INTERVAL,
+      },
+    },
+  },
+  methods: {
+    _getLastDateTime() {
+      return this.db.get("dateTime");
+    },
+    _setDateTime(dateTime) {
+      this.db.set("dateTime", dateTime);
+    },
+    async processEvent({
+      params, dateTime, func,
+    }) {
+      const items = this.testmonitor.paginate({
+        fn: func,
+        params,
+      });
```

Comment:
I would advise not to paginate in this function which role should only be as the name suggests. So there is a number of examples you can take a look at that you can use as a reference ([activecampaign](https://github.com/PipedreamHQ/pipedream/blob/master/components/activecampaign/sources/common/base.mjs#L21-L24), [amara](https://github.com/PipedreamHQ/pipedream/blob/master/components/amara/sources/common.mjs#L112-L115), [aweber](https://github.com/PipedreamHQ/pipedream/blob/master/components/aweber/sources/common.mjs#L72-L75) etc)